gdpr/impl.go

```diff
@@ -110,7 +117,7 @@ func (p *permissionsImpl) parseVendor(ctx context.Context, vendorID uint16, cons
 		return
 	}
 
-	vendorList, err := p.fetchVendorList(ctx, parsedConsent.VendorListVersion())
+	vendorList, err := p.fetchVendorList[parsedConsent.Version()-1](ctx, parsedConsent.VendorListVersion())
```

Comment:
This should be safe. If version is anything other than 1 or 2, we would get an error from the library and not run this code.. right?

Reply:
Actually once TCF 3.0 comes out, that definitely won't be true. We should do a check on this side.